net/nimble/host/src/ble_gap.c

```diff
@@ -1772,6 +1792,38 @@ ble_gap_adv_type(const struct ble_gap_adv_params *adv_params)
     }
 }
 
+#if MYNEWT_VAL(BLE_EXT_ADV)
+static uint16_t
+ble_gap_ext_adv_prop(const struct ble_gap_adv_params *adv_params)
+{
+    uint16_t prop = 0;
+
+    switch (adv_params->conn_mode) {
+    case BLE_GAP_CONN_MODE_NON:
+        if (adv_params->disc_mode == BLE_GAP_DISC_MODE_NON) {
+            prop |= BLE_HCI_LE_SET_EXT_ADV_PROP_SCANNABLE;
+        }
+        break;
+    case BLE_GAP_CONN_MODE_UND:
+        prop |= BLE_HCI_LE_SET_EXT_ADV_PROP_CONNECTABLE;
+        break;
+    case BLE_GAP_CONN_MODE_DIR:
+        prop |= BLE_HCI_LE_SET_EXT_ADV_PROP_CONNECTABLE;
+        prop |= BLE_HCI_LE_SET_EXT_ADV_PROP_DIRECTED;
+
+        if (adv_params->high_duty_cycle) {
+            prop = BLE_HCI_LE_SET_EXT_ADV_PROP_HD_DIRECTED;
```

Comment:
this bit should not be used if extended advertising pdu types are used (and as I understand this function is used to convert properties only in this case)